CmdStarter.Tests/Attributes.cs

```diff
@@ -0,0 +1,141 @@
+        [TestCase<PartlyHiddenCommandB>(PartlyHiddenCommandB.NAME_OF_OPTION, PartlyHiddenCommandB.OPTION_IS_HIDDEN, PartlyHiddenCommandB.NAME_OF_PARAMETER, PartlyHiddenCommandB.PARAMETER_IS_HIDDEN, PartlyHiddenCommandB.COMMAND_IS_HIDDEN)]
+        [TestCase<PartlyHiddenCommandC>(PartlyHiddenCommandC.NAME_OF_OPTION, PartlyHiddenCommandC.OPTION_IS_HIDDEN, PartlyHiddenCommandC.NAME_OF_PARAMETER, PartlyHiddenCommandC.PARAMETER_IS_HIDDEN, PartlyHiddenCommandC.COMMAND_IS_HIDDEN)]
+        public void EnsureIsHiddenAttribute<CommandType>(string optionName, bool isOptionHidden, string argumentName, bool isArgumentHidden, bool commandHidden)
+            where CommandType : StarterCommand
+        {
+            starter.Namespaces = starter.Namespaces.Add(typeof(CommandType).Namespace!);
+            starter.InstantiateCommands();
+
+            //Test command
+            var command = starter.FindCommand<CommandType>() as CommandType;
+            Assert.That(command, Is.Not.Null);
+            Assert.That(command.IsHidden, Is.EqualTo(commandHidden));
+
+            //Test argument
+            var argument = command.Arguments.FirstOrDefault(argument => argument.Name == argumentName);
+            Assert.That(argument, Is.Not.Null);
+            Assert.That(argument.IsHidden, Is.EqualTo(isArgumentHidden));
+
+            //Test option
+            var option = command.Options.FirstOrDefault(option => option.Name == optionName);
+            Assert.That(option, Is.Not.Null);
+            Assert.That(option.IsHidden, Is.EqualTo(isOptionHidden));
+        }
+
+        [TestCase<NoAutoComplete>(NoAutoComplete.OPTION_NAME, NoAutoComplete.ARGUMENT_NAME)]
+        [TestCase<SingleSingleAutoComplete>(SingleSingleAutoComplete.OPTION_NAME, SingleSingleAutoComplete.ARGUMENT_NAME)]
+        [TestCase<SingleMultiAutoComplete>(SingleMultiAutoComplete.OPTION_NAME, SingleMultiAutoComplete.ARGUMENT_NAME)]
+        [TestCase<MultiSingleAutoComplete>(MultiSingleAutoComplete.OPTION_NAME, MultiSingleAutoComplete.ARGUMENT_NAME)]
+        [TestCase<MultiMultiAutoComplete>(MultiMultiAutoComplete.OPTION_NAME, MultiMultiAutoComplete.ARGUMENT_NAME)]
+        [TestCase<LonelyEnumAutoComplete>(LonelyEnumAutoComplete.OPTION_NAME, LonelyEnumAutoComplete.ARGUMENT_NAME)]
+        [TestCase<LonelyFeederAutoCompletion>(LonelyFeederAutoCompletion.OPTION_NAME, LonelyFeederAutoCompletion.ARGUMENT_NAME)]
+        [TestCase<FactoryAutoComplete>(FactoryAutoComplete.OPTION_NAME, FactoryAutoComplete.ARGUMENT_NAME)]
+        [TestCase<FactoryEnumAutoComplete>(FactoryEnumAutoComplete.OPTION_NAME, FactoryEnumAutoComplete.ARGUMENT_NAME)]
+        [TestCase<FactoryFeederAutoCompletion>(FactoryFeederAutoCompletion.OPTION_NAME, FactoryFeederAutoCompletion.ARGUMENT_NAME)]
+        public void EnusreAutoCompleteAttribute<CommandType>(string optionName, string argumentName)
```

Comment:
Two small things:
- Up to now, I named tests that throws exceptions `Throws...`
- I see the tests as being an `if` condition: passing, failing. From this point of view, the naming convention I've taken for tests was a verb at the 3rd singular person (like throws). This is the naming convention for methods that returns a boolean. In this case, it doesn't return a boolean, but its states are boolean: only two. Your thoughts?